plugins/inputs/snmp/snmp.go

```diff
@@ -460,6 +462,21 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
 					}, idx)
 				}
 
+				// snmptranslate table field value here
+				if f.Translate {
+					switch ent.Value.(interface{}).(type) {
+					case string:
+						{
+							entOid := ent.Value.(string)
+							_, _, oidText, _, err := SnmpTranslate(entOid)
+							if err == nil {
+								// If no error translating, the original value for ent.Value should be replaced
+								ent.Value = oidText
+							}
+						}
+					}
```

Comment:
I think this is overly complex. How about
```suggestion
					if entOid, ok := ent.Value.(string); ok {
						_, _, oidText, _, err := SnmpTranslate(entOid)
						if err == nil {
							// If no error translating, the original value for ent.Value should be replaced
							ent.Value = oidText
						}
					}
```

Reply:
Your'e right, the switch is redundant here.